src/__tests__/labs/requests/NewLabRequest.test.tsx

```diff
@@ -13,205 +13,169 @@ import NewLabRequest from '../../../labs/requests/NewLabRequest'
-      const typeInputBox = wrapper.find(TextInputWithLabelFormGroup)
+      setup()
 
-      expect(typeInputBox).toBeDefined()
-      expect(typeInputBox.prop('label')).toEqual('labs.lab.type')
-      expect(typeInputBox.prop('isRequired')).toBeTruthy()
-      expect(typeInputBox.prop('isEditable')).toBeTruthy()
+      expect(screen.getByText(/labs\.lab\.type/i)).toHaveAttribute(
+        'title',
+        'This is a required input',
+      )
+      expect(screen.getByLabelText(/labs\.lab\.type/i)).toBeInTheDocument()
+      expect(screen.getByLabelText(/labs\.lab\.type/i)).not.toBeDisabled()
     })
 
     it('should render a notes text field', async () => {
-      const { wrapper } = await setup()
-      const notesTextField = wrapper.find(TextFieldWithLabelFormGroup)
+      setup()
 
-      expect(notesTextField).toBeDefined()
-      expect(notesTextField.prop('label')).toEqual('labs.lab.notes')
-      expect(notesTextField.prop('isRequired')).toBeFalsy()
-      expect(notesTextField.prop('isEditable')).toBeTruthy()
+      expect(screen.getByLabelText(/labs\.lab\.notes/i)).not.toBeDisabled()
+      expect(screen.getByText(/labs\.lab\.notes/i)).not.toHaveAttribute(
+        'title',
+        'This is a required input',
+      )
     })
 
     it('should render a visit select', async () => {
-      const { wrapper } = await setup()
-      const visitSelect = wrapper.find(SelectWithLabelFormGroup)
+      setup()
 
-      expect(visitSelect).toBeDefined()
-      expect(visitSelect.prop('label') as string).toEqual('patient.visit')
-      expect(visitSelect.prop('isRequired')).toBeFalsy()
-      expect(visitSelect.prop('defaultSelected')).toEqual([])
+      const selectLabel = screen.getByText(/patient\.visit/i)
+      const selectInput = screen.getByPlaceholderText('-- Choose --')
+
+      expect(selectInput).toBeInTheDocument()
+      expect(selectInput).toHaveDisplayValue([''])
+      expect(selectLabel).toBeInTheDocument()
+      expect(selectLabel).not.toHaveAttribute('title', 'This is a required input')
     })
 
-    it('should render a save button', async () => {
-      const { wrapper } = await setup()
-      const saveButton = wrapper.find(Button).at(0)
-      expect(saveButton).toBeDefined()
-      expect(saveButton.text().trim()).toEqual('labs.requests.new')
+    it('should render a save button', () => {
+      setup()
+
+      expect(screen.getByRole('button', { name: /labs\.requests\.new/i })).toBeInTheDocument()
     })
 
-    it('should render a cancel button', async () => {
-      const { wrapper } = await setup()
-      const cancelButton = wrapper.find(Button).at(1)
-      expect(cancelButton).toBeDefined()
-      expect(cancelButton.text().trim()).toEqual('actions.cancel')
+    it('should render a cancel button', () => {
+      setup()
+
+      expect(screen.getByRole('button', { name: /actions\.cancel/i })).toBeInTheDocument()
     })
 
     it('should clear visit when patient is changed', async () => {
```

Comment:
I made a change in Dashboard.tsx to test something else. I eventually reverted to code back to what it is currently. Rerunning tests failed 4 tests including this one.

![image](https://user-images.githubusercontent.com/5468546/103902893-d5e15c00-5103-11eb-880e-75fbe6b7f77c.png)

Reply:
a rerun of all tests passed so not sure. Perhaps it was just a temporary failure.

Reply:
> a rerun of all tests passed so not sure. Perhaps it was just a temporary failure.

Not ideal, we definitely want them to be consistent and stable. We can look more into these ones and what changes you had tested 😄

Reply:
@nobrayner

Reply:
https://github.com/HospitalRun/hospitalrun-frontend/pull/2516#discussion_r554181501